### PR TITLE
Site Settings: Remove placeholder and related site existence checks

### PR DESCRIPTION
**Pull request by tyxla**
Some of our site settings still check for site existence. However, these checks are no longer necessary, as we won't be displaying the steps if the site isn't loaded yet. This also allows us to remove the placeholder that we displayed in that case.

This likely doesn't remove all the site existence checks, but hey - we have to start somewhere 😉 

#### Changes proposed in this Pull Request

* Remove the unnecessary site existence checks in site settings.
* Delete the placeholder component.
* Janitorial: Remove unnecessary prettier pragmas.
* Janitorial: Sort imports.
* Janitorial: Simplify functional components

#### Testing instructions

* Checkout this branch.
* Test all affected pages and make sure there are no regressions:
  * Guided transfer: `/settings/export/guided/:site` for Business .com sites.
  * Jetpack site disconnection flow.
  * General settings.
  * Export settings.
  * Import settings.
  * Discussion settings.
  * Security settings.
  * Writing settings.
* Verify the placeholder component is indeed no longer used.

### CODE REVIEW

File: client/my-sites/site-settings/disconnect-site/too-expensive.jsx [hunk 1]
```diff
@@ -10,19 +9,17 @@ import { localize } from 'i18n-calypso';
...
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
 
 const TooExpensive = ( { confirmHref, planPurchaseId, siteId, siteSlug, translate } ) => {
-	if ( ! siteSlug || ! planPurchaseId ) {
+	if ( ! planPurchaseId ) {
 		return (
 			<div>
 				<QuerySitePlans siteId={ siteId } />
-				<Placeholder />
```
Comment (jsnajdr) [inline client/my-sites/site-settings/disconnect-site/too-expensive.jsx]: The placeholder should still be displayed while the site plans are loading. Unlike the site itself, the plans are not guaranteed to be available and need to load first.

**Review by jsnajdr — DISMISSED** [review]
There's one place where the placeholder is still useful: when loading site plans in `/settings/disconnect-site/too-expensive/site.blog`. Let me know what you think 🙂

File: client/my-sites/site-settings/disconnect-site/too-expensive.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/site-settings/disconnect-site/too-expensive.jsx]: This happens in such a rare case that I'm more than fine to remove the placeholder there - I did remove it intentionally. It is not beneficial right there IMHO, and just displaying nothing is more than fine. How do you feel about that?

**Comment by tyxla** [comment]
@jsnajdr this was intentional on my side 😉 Removing the placeholder from there does no harm IMHO, as displaying the placeholder on that page isn't much different from displaying nothing while the plans are loading. Let me know if you're good to remove it (considering it's the last instance of the placeholder).

**Review by jsnajdr — APPROVED** [review]
If the `Placeholder` removal was intentional, then it's OK. I agree it's not worth keeping the component just for one use case.

**Pull request merged**